jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MinimumCycleMeanAlgorithm.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (C) Copyright 2020-2020, by Semen Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.interfaces;
+
+import org.jgrapht.GraphPath;
+
+/**
+ * The algorithm for finding minimum cycle mean in a graph.
+ *
+ * <p>
+ * Consider a cycle $C$ in a graph. The mean of cycle $C$ is defined as
+ * $\lambda (C)=\frac{w(C)}{|C|}$, where $w(C)$ is the total weight of $C$ and
+ * $|C|$ is the length of $C$.
+ *
+ * @param <V> graph vertex type
+ * @param <E> graph edge type
+ */
+public interface MinimumCycleMeanAlgorithm<V, E> {
```

Comment:
Are there alternative algorithms that compute a minimumCycleMean? If not, do we really need an interface? (I had never heard about a Minimum Cycle Mean before seeing this PR, so please excuse my ignorance).

Reply:
The idea to implement this algorithm was derived from the project`s wiki page. There are two other algorithms mentioned there, in particular the Karp`s algorithm and the Hartman-Orlin`s algorithm. Furthermore, in the article several more algorithms are mentioned.

I decided to start with this one because it is reported to be the fastest one. Never the less its complexity is exponential. The Karp`s and the Hartman-Orlin`s algorithms have polynomial complexities in terms of number of vertices and number of edges. So there is a theoretical improvement there.

Does it make sense to you if we had additionally an implementation of a polynomial algorithm for this problem even though it possibly not going to be faster in most of the cases? If so, I could implement it in this PR with a corresponding performance comparison.

Reply:
Potentially we can implement a polynomial algorithm for this problem, so this interface makes sense to me. However, I would prefer not to add it in this PR as it will complicated the review process

Reply:
ok